android/quest/src/main/java/org/smartregister/fhircore/quest/ui/appsetting/AppSettingViewModel.kt

```diff
@@ -129,15 +132,20 @@ constructor(
                 if (bundleEntryComponent.resource is Binary) {
                   val binary = bundleEntryComponent.resource as Binary
                   binary.data.decodeToString().decodeBase64()?.string(StandardCharsets.UTF_8)?.let {
-                    val config =
-                      it.tryDecodeJson<RegisterConfiguration>()
-                        ?: it.tryDecodeJson<ProfileConfiguration>()
-
-                    when (config) {
-                      is RegisterConfiguration ->
-                        config.fhirResource.dependentResourceTypes(patientRelatedResourceTypes)
-                      is ProfileConfiguration ->
-                        config.fhirResource.dependentResourceTypes(patientRelatedResourceTypes)
+                    val registerConfig = it.tryDecodeJson<RegisterConfiguration>()
+                    if (registerConfig != null) {
+                      if (registerConfig.configType == ConfigType.Profile.name) {
+                        val profileConfig = it.tryDecodeJson<ProfileConfiguration>()
+                        profileConfig
+                          ?.fhirResource
+                          ?.dependentResourceTypes(
+                            patientRelatedResourceTypes,
+                          )
+                      } else {
+                        registerConfig.fhirResource.dependentResourceTypes(
+                          patientRelatedResourceTypes,
+                        )
+                      }
```

Comment:
because we are ignoring unknown keys, as it was written it always decodes as RegisterConfiguration

Reply:
with the new implementation looks like we might experience a crash if registerConfig was actually ever null

Reply:
maybe, it would be a pre-existing issue since that's the same as config getting set to null in the previous version, I think we have a test for that, but it might be passing b/c of mocks